contracts/src/Tracker.sol

```diff
@@ -58,21 +60,34 @@ contract Tracker is EIP712 {
         string memory description,
         uint256 minimumCheckIns,
         uint256 startTimestamp,
+        uint256 joinDueTimestamp,
         uint256 endTimestamp,
         address donateDestination,
         uint256 stake
     ) public {
-        //require(endTimestamp > startTimestamp, "End timestamp must be greater than start timestamp");
-        challenges[++challengeCounter] =
-            Challenge(verifier, minimumCheckIns, startTimestamp, endTimestamp, donateDestination, stake, 0, false);
-        emit Register(challengeCounter, verifier, description, startTimestamp, endTimestamp, minimumCheckIns);
+        require(endTimestamp > startTimestamp, "end timestamp must be greater than start timestamp");
+        require(joinDueTimestamp <= endTimestamp, "should not allow joining time larger than challenge ending time");
+        challenges[++challengeCounter] = Challenge(
+            verifier,
+            minimumCheckIns,
+            startTimestamp,
+            joinDueTimestamp,
+            endTimestamp,
+            donateDestination,
+            stake,
+            0,
+            false
+        );
+        emit Register(
+            challengeCounter, verifier, description, startTimestamp, joinDueTimestamp, endTimestamp, minimumCheckIns
```

Comment:
```suggestion
            challengeCounter, verifier, extraData, startTimestamp, joinDueTimestamp, endTimestamp, minimumCheckIns
```

Also taking this opportunity to name it something more general and less opinionated. so we can put name, description and any additional info in as `extraData`